companion/src/firmwares/opentx/opentxeeprom.cpp

```diff
@@ -1386,33 +1386,31 @@ class InputField: public TransformedField {
         }
         else {
           _curveMode = false;
-          _curveParam = smallGvarToEEPROM(expo.curve.value);
+          _curveParam = smallGvarImport(expo.curve.value);
         }
       }
     }
 
     virtual void afterImport()
     {
-      if (IS_STM32(board)) {
-        if (version < 216) {
-          if (expo.mode) {
-            expo.srcRaw = RawSource(SOURCE_TYPE_STICK, expo.chn);
-          }
+      if (IS_STM32(board) && version < 216) {
+        if (expo.mode) {
+          expo.srcRaw = RawSource(SOURCE_TYPE_STICK, expo.chn);
         }
       }
       else if (expo.mode) {
         expo.srcRaw = RawSource(SOURCE_TYPE_STICK, expo.chn);
       }
 
-      expo.weight = smallGvarToC9x(_weight);
+      expo.weight = smallGvarExport(_weight);
 
-      if (IS_STM32(board) && version >= 216) {
-        expo.offset = smallGvarToC9x(_offset);
+      if ((IS_STM32(board) && version >= 216) || (IS_ARM(board) && version >= 218)) {
+        expo.offset = smallGvarExport(_offset);
       }
 
-      if (!IS_ARM(board) || (!IS_STM32(board) && version < 218) || version < 216) {
```

Comment:
You are right for IS_STM32
But I think the !IS_ARM(board) should remain as it was (no test for 218), as I did above
And yes this would be the same test in import and export